src/main/java/org/opensearch/sdk/handlers/UpdateSettingsRequestHandler.java

```diff
@@ -99,9 +99,9 @@ public AcknowledgedResponse handleUpdateSettingsRequest(UpdateSettingsRequest up
                 case Version:
                     Consumer<Version> versionConsumer = (Consumer<Version>) this.settingUpdateConsumers.get(componentSetting);
                     versionConsumer.accept((Version) data);
-                default:
-                    throw new UnsupportedOperationException("Setting Update Consumer type does not exist and is not handled here");
             }
+        } catch (UnsupportedOperationException e) {
+            throw new UnsupportedOperationException("Setting Update Consumer type does not exist and is not handled here");
```

Comment:
I'm used to _always_ including a `default` in switch/case statements, even when it will never happen.  

I'm not sure I understand the reason for this change, or what it fixes. Could you explain the benefit of your change here?

Reply:
these changes were made to remove this: An exception is intercepted when an exception is not generated in the org.opensearch.sdk.handlers settings.Update in UpdateSettingsRequestHandler.java :[line 105]

Reply:
> these changes were made to remove this: An exception is intercepted when an exception is not generated in the org.opensearch.sdk.handlers settings.Update in UpdateSettingsRequestHandler.java :[line 105]

OK, this is one of those cases where with the existing code, the exception will never be thrown (thus spotbugs complains).  But if someone makes code changes elsewhere and doesn't handle them here, this exception will alert the developer that they forgot something.

So there's no need to change anything here.  If there's a way to add a configuration to suppress spotbugs on this issue, that'd be preferred.

Reply:
is it possible to add an additional catch(Runtimeexception e) to remove this warning because the @SuppressWarning("all") annotation or  unchecked, unused or other options do not work with this annotation